datanode/sqlsubscribers/order.go

```diff
@@ -44,17 +51,43 @@ func NewOrder(store OrderStore) *Order {
 }
 
 func (os *Order) Types() []events.Type {
-	return []events.Type{events.OrderEvent}
+	return []events.Type{
+		events.OrderEvent,
+		events.ExpiredOrdersEvent,
+	}
 }
 
 func (os *Order) Push(ctx context.Context, evt events.Event) error {
-	return os.consume(evt.(OrderEvent), evt.Sequence())
+	switch evt.Type() {
+	case events.OrderEvent:
+		return os.consume(evt.(OrderEvent), evt.Sequence())
+	case events.ExpiredOrdersEvent:
+		return os.expired(ctx, evt.(ExpiredOrdersEvent), evt.Sequence())
+	}
+	return nil
 }
 
 func (os *Order) Flush(ctx context.Context) error {
 	return os.store.Flush(ctx)
 }
 
+func (os *Order) expired(ctx context.Context, eo ExpiredOrdersEvent, seqNum uint64) error {
+	orders, err := os.store.GetByMarketAndID(ctx, eo.MarketID(), eo.OrderIDs())
+	if err != nil {
+		return err
+	}
+	for _, o := range orders {
```

Comment:
Won't tyou have the wrong TxHash here? You'll need to set it sa well, see the consume function.

Reply:
Wrong is such a strong word, but it would indeed be, erm, less correct. Updated